### ISSUE TITLE
Electron binaries 

### ISSUE THREAD

**Issue by marcopiraccini** [tojocky/node-printer#113]
It would be good to have binaries for Electron builded for Windows / Linux and Darwin + released on github
If you agree, I can prepare a PR for that

**Comment by samt** [comment]
Using this with electron (which is a really awesome place to use this module by the way) is a really pain because of this reason. Pre-build binaries for this would be awesome.

**Comment by antoinepairet** [comment]
See also #104

**Comment by shoaibmerchant** [comment]
+1

**Comment by xiongsongsong** [comment]
hello ,can send a pre-compiled files to my mailbox it?  I currently use windows system~
xiong_song@126.com
thanks! @marcopiraccini

**Comment by xiongsongsong** [comment]
I really hope to provide pre-compiled version.
Because I tested many times in a virtual machine, or compile unsuccessful, too much trouble.

<img width="915" alt="screen shot 2016-05-24 at 8 50 27 am" src="https://cloud.githubusercontent.com/assets/342509/15488802/b31eeb20-218c-11e6-9729-b931c591e47f.png">

**Comment by xiongsongsong** [comment]
@tojocky

**Comment by xiongsongsong** [comment]
Like Electron, offer versions for each platform, you know, going to the user's computer is not easy to compile node-printer.

![image](https://cloud.githubusercontent.com/assets/342509/15526136/206d4084-2262-11e6-9cda-8f01fff973a9.png)

**Comment by tojocky** [comment]
Could you please prepare a PR?

**Comment by xiongsongsong** [comment]
@marcopiraccini @tojocky I'm sorry, I'm a web developer engineer, not C language, very sorry. 😞

**Comment by vasumahesh1** [comment]
Did a test run and was able to generate release and publish:
https://github.com/vasumahesh1/node-printer/releases/tag/0.2.1

I hope this can work. Will submit a PR soon.

**Comment by xiongsongsong** [comment]
@vasumahesh1  thanks!!! 👍👍 

**Already perfect run it!**

### Windows 7
![image](https://cloud.githubusercontent.com/assets/342509/15542073/c2411386-22c1-11e6-91fa-9490c7702a77.png)

### Windows 10
![image](https://cloud.githubusercontent.com/assets/342509/15542428/53cff988-22c3-11e6-9873-06efd1480164.png)

**Comment by xiongsongsong** [comment]
This is a project simply can not run, do not write on the windows compatible.

**Comment by vasumahesh1** [comment]
@xiongsongsong You can download the releases from my repo.

https://github.com/vasumahesh1/node-printer/releases/tag/0.2.1

I have tested the 32bit & 64bit Electron releases. Both are working fine.

Haven't tested the Win32 builds. But pretty sure they will work.
To install:
```sh
# For electron builds
npm install vasumahesh1/node-printer --runtime=electron --target=0.36.1 --target_arch=x64
npm install vasumahesh1/node-printer --runtime=electron --target=0.36.1 --target_arch=ia32

# For regular builds
npm install vasumahesh1/node-printer --target_arch=ia32
npm install vasumahesh1/node-printer --target_arch=x64
```

So far I am using only 0.36.1 electron version. If you want another version to be built let me know.

**Comment by ibuger** [comment]
@vasumahesh1,I use your "node_printer-v0.2.1-node-v47-win32-ia32.tar.gz" , when i use the method of printDirect,it tip me "Uncaught TypeError: Expected 5 arguments",how can i call this method?

**Comment by Lithas** [comment]
@vasumahesh1 Any chance you could do 1.2.6?

**Comment by vasumahesh1** [comment]
@Lithas Yeah I look into it... been busy with work. Will get back when they are up.

**Comment by vasumahesh1** [comment]
@Lithas Hey check now, I built it. You might wanna test it as I don't have 1.2.x electron with me. Let me know if it doesn't work.

**Comment by Lithas** [comment]
@vasumahesh1  Thanks for being so quick! I'll try to test it out tonight and let you know for sure

**Comment by zimt28** [comment]
Will these builds be released officially? I'd love one for Mac, as it's a pain having to install XCode just for one package /:

**Comment by seiyria** [comment]
Would be nice to have these - this library is the most painful of all that I use because of the lack of prebuilt binaries. Makes it a pain to work with :(
